test/cases/upgrade8to9.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+set -euxo pipefail
+
+# Get OS data.
+source /usr/libexec/osbuild-composer-test/set-env-variables.sh
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m[$(date -Isecond)] ${1}\033[0m"
+}
+
+# Wait for VM to start
+function wait_for_vm {
+    INSTANCE_ADDRESS=192.168.100.50
+    SSH_OPTIONS=(-o StrictHostKeyChecking=no -o ConnectTimeout=5)
+    NUM_LOOPS=60
+    for _ in $(seq 0 "$NUM_LOOPS"); do
+        if eval 'sudo ssh "${SSH_OPTIONS[@]}" -q -i "${SSH_KEY}" root@"$INSTANCE_ADDRESS" exit'; then
+            echo "Test VM is up."
+            break
+        else
+            echo "Test VM is not ready yet."
+        fi
+        sleep 10
+    done
+}
+
+# Start libvirtd and test it.
+greenprint "🚀 Starting libvirt daemon"
+sudo systemctl start libvirtd
+sudo virsh list --all > /dev/null
+
+# Set a customized dnsmasq configuration for libvirt so we always get the
+# same address on bootup.
+sudo tee /tmp/integration.xml > /dev/null << EOF
+<network>
+  <name>integration</name>
+  <uuid>1c8fe98c-b53a-4ca4-bbdb-deb0f26b3579</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='integration' stp='on' delay='0'/>
+  <mac address='52:54:00:36:46:ef'/>
+  <ip address='192.168.100.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.100.2' end='192.168.100.254'/>
+      <host mac='34:49:22:B0:83:30' name='vm' ip='192.168.100.50'/>
+    </dhcp>
+  </ip>
+</network>
```

Comment:
From what I can see all values here are hard-coded so that could go into the data directory instead of a here document.

Reply:
Right, I'll just move it there.

Reply:
Moved.